From aea67c229bdbfbb24fb80766f9c1979599e75397 Mon Sep 17 00:00:00 2001
From: Nikita Travkin <nikita@trvn.ru>
Date: Wed, 24 Jul 2019 19:43:31 +0500
Subject: [PATCH 097/210] arm64: dts: qcom: msm8916-longcheer-l8150: Add sound
 and modem (v8)

SOUND:
v2: Squash audio jack / headset key detection
v3: Rename dai-link nodes, add labels
v4: Fix jack detection type

MODEM:
v2: Set lpass@7708000 to status = "disabled"
v3: Add q6asm (MultiMedia Playback/Capture) routes
v4: Add CS-Voice link, add MultiMedia4 for compressed playback
v5: Use common msm8916-modem.dtsi
v6: Setup memshare / GPS memory
v7: Venus memory no longer needs to be relocated

Co-developed-by: Stephan Gerhold <stephan@gerhold.net>
Signed-off-by: Nikita Travkin <nikita@trvn.ru>

v8: Combine sound and modem commits and drop extra -modem.dts
    The no-modem variant is now patched dynamically by lk2nd instead.

Signed-off-by: Stephan Gerhold <stephan@gerhold.net>
---
 .../boot/dts/qcom/msm8916-longcheer-l8150.dts | 41 ++++++++++++++++++-
 1 file changed, 40 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/qcom/msm8916-longcheer-l8150.dts b/arch/arm64/boot/dts/qcom/msm8916-longcheer-l8150.dts
index 438c26e9f15..e9288236c74 100644
--- a/arch/arm64/boot/dts/qcom/msm8916-longcheer-l8150.dts
+++ b/arch/arm64/boot/dts/qcom/msm8916-longcheer-l8150.dts
@@ -3,11 +3,14 @@
 /dts-v1/;
 
 #include "msm8916-pm8916.dtsi"
+#include "msm8916-modem.dtsi"
+
 #include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/input/input.h>
 #include <dt-bindings/interrupt-controller/irq.h>
 #include <dt-bindings/leds/common.h>
 
+/delete-node/ &mpss_mem;
 /delete-node/ &wcnss_mem;
 
 / {
@@ -23,12 +26,33 @@ chosen {
 		stdout-path = "serial0";
 	};
 
+	/*
+	 * For some reason, the wcnss firmware is not relocatable, it must be
+	 * loaded at 0x8b600000. Unfortunately this overlaps with the memory
+	 * region required for the modem (0x86800000-0x8b800000).
+	 *
+	 * On the downstream kernel everything somehow works even with the
+	 * overlap (possibly because wcnss is loaded after the modem
+	 * and the modem never accessed its last 2 MB memory...?)
+	 *
+	 * Fortunately at least the modem is relocatable (aligned to 0x400000),
+	 * so put it after everything else and keep the original region empty.
+	 */
 	reserved-memory {
-		/* wcnss.mdt is not relocatable, it must be loaded at 0x8b600000 */
 		wcnss_mem: wcnss@8b600000 {
 			reg = <0x0 0x8b600000 0x0 0x600000>;
 			no-map;
 		};
+
+		mpss_mem: mpss@8ec00000 {
+			reg = <0x0 0x8ec00000 0x0 0x5000000>;
+			no-map;
+		};
+
+		gps_mem: gps@93c00000 {
+			reg = <0x0 0x93c00000 0x0 0x200000>;
+			no-map;
+		};
 	};
 
 	gpio-keys {
@@ -248,6 +272,14 @@ &sdhc_2 {
 	non-removable;
 };
 
+&sound {
+	status = "okay";
+	audio-routing =
+		"AMIC1", "MIC BIAS Internal1",
+		"AMIC2", "MIC BIAS Internal2",
+		"AMIC3", "MIC BIAS Internal3";
+};
+
 &usb {
 	status = "okay";
 	dr_mode = "peripheral";
@@ -258,6 +290,13 @@ &usb_hs_phy {
 	extcon = <&pm8916_usbin>;
 };
 
+&wcd_codec {
+	qcom,micbias-lvl = <2800>;
+	qcom,mbhc-vthreshold-low = <75 150 237 450 500>;
+	qcom,mbhc-vthreshold-high = <75 150 237 450 500>;
+	qcom,hphl-jack-type-normally-open;
+};
+
 &smd_rpm_regulators {
 	vdd_l1_l2_l3-supply = <&pm8916_s3>;
 	vdd_l4_l5_l6-supply = <&pm8916_s4>;
-- 
2.34.1

